src/typeInference/annotateBinaryOperator.ts

```diff
@@ -34,82 +35,130 @@ export function annotateAddOp(
 			type: BINOP_LOOKUP[left.type + right.type * 1000] || ValueType.XSDECIMAL,
 			mult: left.mult,
 		};
-		ast.push(['type', type]);
+		insertAttribute(ast, type);
 		return type;
 	}
 
 	if (
 		left.type === ValueType.XSYEARMONTHDURATION &&
 		right.type === ValueType.XSYEARMONTHDURATION
 	) {
-		ast.push(['type', { type: ValueType.XSYEARMONTHDURATION, mult: left.mult }]);
-		return { type: ValueType.XSYEARMONTHDURATION, mult: left.mult };
+		const yearMonthDurationSequenceType = {
```

Comment:
Good idea to move this into a separate variable

Reply:
Well every bit counts